src/visidentifier/visprotocol.cpp

```diff
@@ -0,0 +1,238 @@
+
+MessageHeader::MessageHeader(const std::string& action, const std::string& requestId)
+    : action(action)
+    , requestId(requestId)
+{
+}
+
+MessageHeader::MessageHeader(const std::string& json)
+{
+
+    try {
+        const auto              result  = json_utils::ParseJson(json);
+        Poco::JSON::Object::Ptr pObject = result.extract<Poco::JSON::Object::Ptr>();
+        if (pObject.isNull())
+            return;
+
+        action    = pObject->getValue<std::string>("action");
+        requestId = pObject->getValue<std::string>("requestId");
+    } catch (const Poco::Exception& e) {
+    }
+}
+
+MessageHeader::MessageHeader(Poco::JSON::Object::Ptr pObject)
+{
+    try {
+        if (pObject.isNull())
+            return;
+
+        action    = pObject->getValue<std::string>("action");
+        requestId = pObject->getValue<std::string>("requestId");
+    } catch (const Poco::Exception& e) {
+    }
+}
+
+json_object MessageHeader::toJSON() const
+{
+    json_object json;
+
+    json.set("action", action);
+    json.set("requestId", requestId);
+
+    return json;
+}
+
+std::string MessageHeader::toString() const
+{
+    std::ostringstream jsonStream;
+    Poco::JSON::Stringifier::stringify(toJSON(), jsonStream);
+
+    return jsonStream.str();
+}
+
+GetRequest::GetRequest(const MessageHeader& messageHeader, const std::string& path)
+    : messageHeader(messageHeader)
+    , path(path)
+{
+}
+
+GetRequest::GetRequest(const std::string& json)
+{
+
+    try {
+        const auto              result  = json_utils::ParseJson(json);
+        Poco::JSON::Object::Ptr pObject = result.extract<Poco::JSON::Object::Ptr>();
+        if (pObject.isNull())
+            return;
+
+        messageHeader = MessageHeader(pObject);
+        path          = pObject->getValue<std::string>("path");
+    } catch (const Poco::Exception& e) {
+    }
+}
+
+json_object SubscribeRequest::toJSON() const
+{
+    auto json = messageHeader.toJSON();
+    json.set("path", path);
+
+    if (filters.empty() == false) { }
+    return json;
+}
+
+GetResponse::GetResponse(const MessageHeader& messageHeader)
+    : messageHeader(messageHeader)
+{
+}
+
+GetResponse::GetResponse(const std::string& json)
+{
+
+    try {
+        const auto              result  = json_utils::ParseJson(json);
+        Poco::JSON::Object::Ptr pObject = result.extract<Poco::JSON::Object::Ptr>();
+        if (pObject.isNull())
+            return;
+
+        messageHeader = MessageHeader(pObject);
+        value         = pObject->getValue<std::string>("value");
+        timestamp     = pObject->getValue<int64_t>("timestamp");
+    } catch (const Poco::Exception& e) {
+    }
+}
+
+json_object GetResponse::toJSON() const
+{
+    auto json = messageHeader.toJSON();
+    json.set("value", value);
+
+    return json;
+}
+
+std::string GetResponse::toString() const
+{
+    std::ostringstream jsonStream;
+    Poco::JSON::Stringifier::stringify(toJSON(), jsonStream);
+
+    return jsonStream.str();
+}
+
```

Comment:
Please split implementation for different classes in sections. Check 

=========

=========
comments in aos_core_lib_cpp